exporter/chronicleexporter/marshal.go

```diff
@@ -224,3 +225,81 @@ func (m *protoMarshaler) constructPayloads(rawLogs map[string][]*api.LogEntry) [
 	}
 	return payloads
 }
+
+func (m *protoMarshaler) MarshalRawLogsForHTTP(ctx context.Context, ld plog.Logs) ([]*api.ImportLogsRequest, error) {
+	rawLogs, err := m.extractRawHTTPLogs(ctx, ld)
+	if err != nil {
+		return nil, fmt.Errorf("extract raw logs: %w", err)
+	}
+
+	return m.constructHTTPPayloads(rawLogs), nil
+}
+
+func (m *protoMarshaler) extractRawHTTPLogs(ctx context.Context, ld plog.Logs) (map[string][]*api.Log, error) {
+	entries := make(map[string][]*api.Log)
+
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceLog := ld.ResourceLogs().At(i)
+		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
+			scopeLog := resourceLog.ScopeLogs().At(j)
+			for k := 0; k < scopeLog.LogRecords().Len(); k++ {
+				logRecord := scopeLog.LogRecords().At(k)
+				rawLog, logType, err := m.processLogRecord(ctx, logRecord, scopeLog, resourceLog)
+				if err != nil {
+					m.teleSettings.Logger.Error("Error processing log record", zap.Error(err))
+					continue
+				}
+
+				if rawLog == "" {
+					continue
+				}
+
+				var timestamp time.Time
+				if logRecord.Timestamp() != 0 {
+					timestamp = logRecord.Timestamp().AsTime()
+				} else {
+					timestamp = logRecord.ObservedTimestamp().AsTime()
+				}
+
+				entry := &api.Log{
+					LogEntryTime:   timestamppb.New(timestamp),
+					CollectionTime: timestamppb.New(logRecord.ObservedTimestamp().AsTime()),
+					Data:           []byte(rawLog),
+				}
+				entries[logType] = append(entries[logType], entry)
+			}
+		}
+	}
+
+	return entries, nil
+}
+
+func buildForwarderString(cfg Config) string {
+	format := "projects/%s/locations/%s/instances/%s/forwarders/%s"
+	return fmt.Sprintf(format, cfg.Project, cfg.Location, cfg.CustomerID, cfg.Forwarder)
+}
+
+func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) []*api.ImportLogsRequest {
+	payloads := make([]*api.ImportLogsRequest, 0, len(rawLogs))
+
+	// TODO logType
```

Comment:
does this logType need to be handled yet or is this another thing that needs clarity?

Reply:
It should be handled now, *however*, there's a bug in the way this is being handled in that's addressed in my other open agent PR. Currently, it won't respect the source parameter for log type, so you need the attribute `log_type` set to `windows_event.security` (that's what Windows Events will set it to)